src/test/regress/expected/multi_subquery_behavioral_analytics.out

```diff
@@ -0,0 +1,997 @@
+  GROUP BY user_id
+) e1 LEFT JOIN LATERAL (
+  SELECT
+    user_id,
+    1 AS use_demo,
+    time AS use_demo_time
+  FROM events_table
+  WHERE
+    user_id = e1.user_id AND
+       event_type IN (11, 21, 31, 41, 51, 61, 71, 81, 91)
+  ORDER BY time
+) e2 ON true LEFT JOIN LATERAL (
+  SELECT
+    user_id,
+    1 AS enter_credit_card,
+    time AS enter_credit_card_time
+  FROM  events_table
+  WHERE
+    user_id = e2.user_id AND
+    event_type IN (12, 22, 32, 42, 52, 62, 72, 82, 92)
+  ORDER BY time
+) e3 ON true LEFT JOIN LATERAL (
+  SELECT
+    1 AS submit_card_info,
+    user_id,
+    time AS enter_credit_card_time
+  FROM  events_table
+  WHERE
+    user_id = e3.user_id AND
+    event_type IN (13, 23, 33, 43, 53, 63, 73, 83, 93)
+  ORDER BY time
+) e4 ON true LEFT JOIN LATERAL (
+  SELECT
+    1 AS see_bought_screen
+  FROM  events_table
+  WHERE
+    user_id = e4.user_id AND
+    event_type IN (14, 24, 34, 44, 54, 64, 74, 84, 94)
+  ORDER BY time
+) e5 ON true
+GROUP BY e1.user_id
+ORDER BY 6 DESC  NULLS LAST, 5 DESC  NULLS LAST, 4 DESC  NULLS LAST, 3 DESC  NULLS LAST, 2 DESC  NULLS LAST, 1
+LIMIT 15;
+ user_id | viewed_homepage | use_demo | entered_credit_card | submit_card_info | see_bought_screen 
+---------+-----------------+----------+---------------------+------------------+-------------------
+      72 |              36 |       36 |                  36 |               36 |                36
+      95 |              12 |       12 |                  12 |               12 |                12
+      82 |               4 |        4 |                   4 |                4 |                 4
+      74 |               3 |        3 |                   3 |                3 |                 3
+      83 |               3 |        3 |                   3 |                3 |                 3
+       6 |               2 |        2 |                   2 |                2 |                 2
+      42 |               1 |        1 |                   1 |                1 |                 1
+       5 |               4 |        4 |                   4 |                4 |                  
+      93 |               4 |        4 |                   4 |                4 |                  
+      51 |               1 |        1 |                   1 |                1 |                  
+      85 |               6 |        6 |                   6 |                  |                  
+      73 |               4 |        4 |                   4 |                  |                  
+       0 |               3 |        3 |                   3 |                  |                  
+      10 |               2 |        2 |                   2 |                  |                  
+      13 |               2 |        2 |                   2 |                  |                  
+(15 rows)
+
+-- Same query without all limitations but uses having() to show only those submitted their credit card info
+SELECT 
```

Comment:
Having such tests is really exciting!

But, could we also add a few failure tests such that join is not equi join or not on the partition key etc?

Also, could you test referring from several levels below for the LATERAL joins. For example, let's say `e5` refers to `e1` (The query probably won't make much sense, but, I'd like to make sure that the codes that we've written for `LATERAL`s works as expected)